Source/Scene/ShadowMap.js

```diff
@@ -385,72 +431,130 @@ define([
         });
 
+        for (var i = 0; i < shadowMap._numberOfPasses; ++i) {
+            shadowMap._passFramebuffers[i] = framebuffer;
+            shadowMap._passStates[i].framebuffer = framebuffer;
+        }
+
         shadowMap._shadowMapTexture = depthStencilTexture;
-        return framebuffer;
+    }
+
+    function createFramebufferCube(shadowMap, context) {
+        var depthRenderbuffer = new Renderbuffer({
+            context : context,
+            width : shadowMap._shadowMapSize,
+            height : shadowMap._shadowMapSize,
+            format : RenderbufferFormat.DEPTH_COMPONENT16
+        });
+
+        var cubeMap = new CubeMap({
+            context : context,
+            width : shadowMap._shadowMapSize,
+            height : shadowMap._shadowMapSize,
+            pixelFormat : PixelFormat.RGBA,
+            pixelDatatype : PixelDatatype.UNSIGNED_BYTE,
+            sampler : createSampler()
+        });
+
+        var faces = [cubeMap.negativeX, cubeMap.negativeY, cubeMap.negativeZ, cubeMap.positiveX, cubeMap.positiveY, cubeMap.positiveZ];
+
+        for (var i = 0; i < 6; ++i) {
+            var framebuffer = new Framebuffer({
+                context : context,
+                depthRenderbuffer : depthRenderbuffer,
+                colorTextures : [faces[i]]
+            });
+            shadowMap._passFramebuffers[i] = framebuffer;
+            shadowMap._passStates[i].framebuffer = framebuffer;
+        }
+
+        shadowMap._shadowMapTexture = cubeMap;
     }
 
     function createFramebuffer(shadowMap, context) {
-        var createFunction = (context.depthTexture) ? createFramebufferDepth : createFramebufferColor;
-        var framebuffer = createFunction(shadowMap, context);
-        shadowMap._framebuffer = framebuffer;
-        shadowMap._passState.framebuffer = framebuffer;
-        for (var i = 0; i < shadowMap._numberOfCascades; ++i) {
-            shadowMap._cascadePassStates[i].framebuffer = framebuffer;
+        if (shadowMap._isPointLight && shadowMap._usesCubeMap) {
+            createFramebufferCube(shadowMap, context);
+        } else if (shadowMap._usesDepthTexture) {
+            createFramebufferDepth(shadowMap, context);
+        } else {
+            createFramebufferColor(shadowMap, context);
         }
     }
 
-    function updateFramebuffer(shadowMap, context) {
-        if (!defined(shadowMap._framebuffer) || (shadowMap._shadowMapTexture.width !== shadowMap._shadowMapSize)) {
+    function checkFramebuffer(shadowMap, context) {
+        // Attempt to make an FBO with only a depth texture. If it fails, fallback to a color texture.
+        if (shadowMap._usesDepthTexture && (shadowMap._passFramebuffers[0].status !== WebGLConstants.FRAMEBUFFER_COMPLETE)) {
+            shadowMap._usesDepthTexture = false;
+            var colorMask = shadowMap._renderState.colorMask;
+            colorMask.red = colorMask.green = colorMask.blue = colorMask.alpha = true;
```

Comment:
It's probably not documented well, but `RenderState`s are read only. This allows us to cache them and make minimal render state changes between draw calls.